scripts/main/sidebar.js

```diff
@@ -205,22 +224,17 @@ sidebar.createStructure.photo = function(data) {
 		}
 	}
 
-	// Only create tags section when the photo is editable
-	if (editable) {
-
-		structure.tags = {
-			title : lychee.locale['PHOTO_TAGS'],
-			type  : sidebar.types.TAGS,
-			value : build.tags(data.tags),
-			editable
-		}
-
-	} else {
-
-		structure.tags = {}
+	// Always create tags section - behaviour for editing
+	//tags handled when contructing the html code for tags
```

Comment:
This begs the question -- are tags supposed to be private or public?

I have no opinion on this so I'm fine with your change but I thought we should check with others...

Reply:
I don't have time for a proper review, but I'd say that if you can see the photo there should be no harm in seeing the tags. For the reverse, I'd say that you should only be able to see tags if you have access to at least one photo with that tag.

Reply:
Agreed with @d7415

Reply:
> For the reverse, I'd say that you should only be able to see tags if you have access to at least one photo with that tag.

@d7415 Could you explain in more detail what you mean?
Let's assume we've got photo 1 (tag A), photo 2 (tag A & B) and photo 3 (no tag). All are pubic.
Do you mean:
(1) Tag A is only displayed at photos to user (in public mode), if at least one of the photos visible to him has tag A? (This would be imho trivial since you can't view tags without viewing a photo <- correct me if I'm wrong).
(2) The section Tags is only shown, if at least one photo visible to the user (in public mode) has a tag -> This would require an additional db query for every photo and backend changes -> I'd vote against this because of (a) simplicity (what's the harm of showing an empty tag section) and (b) performance (I think I read somewhere that there has been recently an attempt to reduce the number of db queries to increase performance)
(3) something completely different

Reply:
I meant 1. This was in a broader sense, in case we list them elsewhere in the future (and in case you'd implemented that here - I still haven't had a change to review it!)

To confirm:

- no concerns with viewing a photo's tags from photo view.
-If a list of tags is going to be used elsewhere (e.g. autocomplete when adding tags) it should only show a tag if the user has access to a photo that uses that tag.

Reply:
Thanks for the explanation. This make sense and is no problem here.